llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6895,8 +6895,9 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
   for (PHINode *PHI : PHIs) {
     const ResultListTy &ResultList = ResultLists[PHI];
 
-    // If using a bitmask, use any value to fill the lookup table holes.
-    Constant *DV = NeedMask ? ResultLists[PHI][0].second : DefaultResults[PHI];
+    // Use any value to fill the lookup table holes.
+    Constant *DV = HolesHaveUndefinedResults ? ResultLists[PHI][0].second
```

Comment:
When the default branch is undefined, we can use a poison value. This can be done as another PR, as it requires modifying the constructor of `SwitchLookupTable`.

Reply:
But I'm not sure if this makes sense. :)

Reply:
I believe this makes sense. For example, we could treat `[1, poison, 3, 4]` as `[1, 2, 3, 4]` or merge `[1, poison, 3, 4]` and `[1, 2, poison, 4]` into `[1, 2, 3, 4]`. If you're considering implementing it, please let me know. :)